plugin/omero_iviewer/views.py

```diff
@@ -276,18 +287,25 @@ def rois_by_plane(request, image_id, the_z, the_t, z_end=None, t_end=None,
     filter.limit = rint(limit)
     params.theFilter = filter
 
-    query = get_query_for_rois_by_plane(the_z, the_t, z_end, t_end,
-                                        load_shapes=True)
+    query = get_query_for_rois_by_plane(the_z, the_t, z_end, t_end)
     rois = query_service.findAllByQuery(query, params, conn.SERVICE_OPTS)
     marshalled = []
     for r in rois:
         encoder = omero_marshal.get_encoder(r.__class__)
         if encoder is not None:
             marshalled.append(encoder.encode(r))
 
+    roi_ids = [r.id.val for r in rois]
+    roi_counts = get_shape_counts(conn, roi_ids)
+
+    for roi in marshalled:
+        if roi['@id'] in roi_counts:
+            roi['shape_count'] = roi_counts[roi['@id']]
+
     # Modify query to only select count() and NOT paginate
     query = get_query_for_rois_by_plane(the_z, the_t, z_end, t_end)
-    query = query.replace("distinct(roi.id)", "count(distinct roi.id)")
+    query = query.replace("select roi", "select count(distinct roi.id)")
```

Comment:
This could be an argument to `get_query_...()`.

Reply:
Done in 8fa122c